src/main/java/subway/line/service/LineService.java

```diff
@@ -0,0 +1,163 @@
+    }
+
+    public List<LineResponse> findAllLines() {
+        return lineRepository.findAll().stream()
+                .map(this::createLineResponse)
+                .collect(Collectors.toList());
+    }
+
+    public LineResponse findOneLine(Long id) {
+        Line line = findOne(id);
+        return createLineResponse(line);
+    }
+
+    @Transactional
+    public void updateLine(Long id, LineRequest lineRequest) {
+        Line line = findOne(id);
+        line.update(lineRequest.getName(), lineRequest.getColor());
+
+        lineRepository.save(line);
+    }
+
+    @Transactional
+    public void deleteLine(Long id) {
+        lineRepository.deleteById(id);
+    }
+
+    private Line findOne(Long id) {
+        return lineRepository.findById(id)
+                .orElseThrow(() -> new IllegalArgumentException("없는 노선 입니다"));
+    }
+
+    @Transactional
+    public void appendSection(Long id, SectionRequest sectionRequest) {
+        Long upStationId = sectionRequest.getUpStationId();
+        Long downStationId = sectionRequest.getDownStationId();
+        Integer distance = sectionRequest.getDistance();
+
+        Line line = findOne(id);
+
+        if (!validateSectionCreation(line, upStationId, downStationId)) {
+            throw new IllegalArgumentException("등록할 수 없는 구간입니다");
+        }
+
+        Station upStation = this.stationService.findOne(upStationId);
+        Station downStation = this.stationService.findOne(downStationId);
+
+        Section section = new Section(distance, upStation, downStation, line);
+        this.sectionRepository.save(section);
+
+        line.appendSection(section);
+        this.lineRepository.save(line);
+    }
+
+    @Transactional
+    public void deleteSection(Long lineId, Long stationId) {
+        Line line = findOne(lineId);
+
+        if(!validateSectionDeletion(line, stationId)) {
+            throw new IllegalArgumentException("제거할 수 없는 구간입니다");
+        }
+
+        line.removeSection();
+        this.lineRepository.save(line);
+    }
+
+    private LineResponse createLineResponse(Line line) {
+        List<StationResponse> stationResponses = line.getSections().stream()
+                .map((s) -> stationService.createStationResponse(s.getUpStation()))
+                .collect(Collectors.toList());
+
+        stationResponses.add(stationService.createStationResponse(line.getDownStation()));
+
+        return LineResponse.builder()
+                .id(line.getId())
+                .name(line.getName())
+                .color(line.getColor())
+                .stations(stationResponses)
+                .build();
+    }
+
+    private Boolean validateSectionCreation(Line line, Long newUpStationId, Long newDownStationId) {
+        if(!equalsDownStation(line, newUpStationId)) {
+            return Boolean.FALSE;
+        }
+
+        if(alreadyInLine(line, newDownStationId)) {
+            return Boolean.FALSE;
+        }
+
+        return Boolean.TRUE;
+    }
```

Comment:
Refernce Type 을 사용하신 이유가 있으실까요?
```suggestion
    private boolean validateSectionCreation(Line line, Long newUpStationId, Long newDownStationId) {
        if(!equalsDownStation(line, newUpStationId)) {
            return false;
        }

        if(alreadyInLine(line, newDownStationId)) {
            return false;
        }

        return true;
    }
```

Reply:
큰 생각 없이 했는데, wrapper 객체는 null의 가능성도 안고 있으니 primitive가 더 확실한 표현인 것 같네요.